### PR TITLE
Log but don't fail on open error

### LINKED ISSUE #9: Open failing should not cause typst-live to fail
On Linux, if `xdg-open` is missing, it breaks `typst-live` with a "file not found" error on this line.

~~~console
% typst-live main.typ
[INFO] Server is listening on http://127.0.0.1:5599/
Error: No such file or directory (os error 2)

Location:
    src/main.rs:43:5
~~~

https://github.com/ItsEthra/typst-live/blob/e1fbaff9259c94d9bb16c2e8184e1293e69406ab/src/main.rs#L43C12-L43C12

If I figure out how to handle errors in Rust, I'll make my own PR. If I don't get around to it, perhaps you'd like to give it a go.

### PR DESCRIPTION
**Pull request by apeschar**
Fixes #9.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -40,7 +40,10 @@ async fn run(state: Arc<ServerState>) -> Result<()> {
         "[INFO] Server is listening on http://{}/",
         server.local_addr()
     );
-    open::that_detached(format!("http://{}", server.local_addr()))?;
+
+    if open::that_detached(format!("http://{}", server.local_addr())).is_err() {
+        println!("[WARN] Could not open the preview in your browser. Try opening it manually...");
```
Comment (ItsEthra) [inline src/main.rs:45]:
```suggestion
        println!("[WARN] Could not open the preview in your browser. Open URL manually: http://{}", server.local_addr());
```

**Review by ItsEthra — CHANGES_REQUESTED** [review]
I think it would be nice if it shows the URL, otherwise looks good.

**Comment by apeschar** [comment]
I think you should be able to enable GitHub Actions/workflows on this repo. The workflow I added would automatically build and run the command to make sure we don't introduce compilation errors with these suggested changes.

**Comment by apeschar** [comment]
Forget what I said, I just hadn't configured the workflow to run on PRs. It's working now.

**Comment by apeschar** [comment]
The Nix workflow does compile and run the project, so any Rust errors would also surface. But it might not report warnings.

**Pull request merged**